resources/wikia/modules/scriptwriter.js

```diff
@@ -24,201 +24,187 @@
 
-		var item, args, callback;
+	/**
+	 * Do an HTML injection
+	 *
+	 * @param {Object}      params
+	 * @param {HTMLElement} params.element  Element to inject HTML to
+	 * @param {string}      params.html     HTML to inject
+	 * @param {Function}    params.callback Function to call when done
+	 */
+	function processItem(params) {
+		log(['processItem', library, params], 'debug', logGroup);
 
-		if (queue.length === 0) {
-			queueCompleted = true;
-			log(['processQueue', 'Queue completed'], 'debug', module);
-			return;
+		if (library === 'postscribe') {
+			win.postscribe(
+				params.element,
+				params.html,
+				{beforeWrite: psBeforeWrite, done: params.callback}
+			);
+		} else {
+			var text = 'document.write(' + JSON.stringify(params.html) + ');';
+			win.ghostwriter(
+				params.element,
+				{
+					insertType: 'append',
+					script: { text: text },
+					done: gwFlushLoadHandlersAndCall(params.callback)
+				}
+			);
 		}
+	}
 
-		item = queue[0];
-		args = item.args;
-		callback = item.callback;
-
-		args.push(function () {
-			log(['Done writing', args], 'debug', module);
-			if (typeof callback === 'function') {
-				log(['Calling done callback', args], 'debug', module);
-				callback();
-				log(['Done callback called', args], 'debug', module);
-			}
-			queue.shift();
-			log(['processQueue', 'Item processed. Going on'], 'debug', module);
-			processQueue();
-		});
+	function processQueue() {
+		log('processQueue', 'debug', logGroup);
 
-		log(['Start writing', args], 'debug', module);
-		impl[item.type].apply(impl, args);
+		lazyQueue.makeQueue(queue, processItem);
+		queue.start();
 	}
 
-	function loadImpl() {
+	function loadLibrary() {
 		var url;
 
-		if (implLoading) {
+		if (loading) {
 			return;
 		}
-		implLoading = true;
+		loading = true;
 
-		log('loadImpl ' + library, 'debug', module);
+		log(['loadLibrary', library], 'debug', logGroup);
 
 		if (library === 'postscribe') {
 			url = '/resources/wikia/libraries/postscribe/postscribe.min.js';
-			impl = implPs;
 
 			// GhostWriter does this safety-guards, but PostScribe doesn't
-			document.open = document.close = function () {
-				return;
-			};
+			doc.open = noop;
+			doc.close = noop;
 
-			if (postscribeLoaded) {
-				log(['loadImpl', 'postscribe already loaded, processing queue'], 'debug', module);
+			if (!!win.postscribe) {
+				log(['loadLibrary', 'postscribe already loaded, processing queue'], 'debug', logGroup);
 				processQueue();
 				return;
 			}
 		} else {
 			url = '/resources/wikia/libraries/ghostwriter/gw.min.js';
-			impl = implGw;
 		}
 
-		if (loader) {
-			loader(url).done(processQueue);
-		} else {
-			require(['wikia.loader'], function (loader) {
-				loader(url).done(processQueue);
-			});
-		}
+		loader(url).done(processQueue);
 	}
 
-	function push(type, args, callback) {
-		log(['push', type, args, callback], 'debug', module);
-
-		queue.push({type: type, args: args, callback: callback});
+	function getElement(elementOrElementId) {
```

Comment:
I wonder if we can have variables declaration on the beginning of the function and assignment later:
```js
function getElement(elementOrElementId) {
	var el;

	if (elementOrElementId && elementOrElementId.getElementsByTagName) {
		return elementOrElementId;
	}

	if (typeof elementOrElementId === 'string') {
		el = doc.getElementById(elementOrElementId);
		if (el) {
			return el;
		}
	}
	
	throw 'wikia.scriptwriter: Not a valid element or element id ' + elementOrElementId;
}
```
But it most likely doesn't make sense - few more bytes of code and the JSHint rule is to have only one `var` per function. It doesn't say that the declaration should be at the very beginning. So, ignore this comment and thanks for reading :japanese_ogre: (I'm jet-lagged :stuck_out_tongue_winking_eye:)